laborator/lab-02/README.md

```diff
@@ -0,0 +1,554 @@
+
+ Pentru desfășurarea acestui laborator vom folosi interfața în linia de comandă.
+
+ Pe parcursul laboratorului, în linia de comandă, vom folosi:
+
+ - comanda `gcc` pe post de linker
+ - `gdb` pentru analiza dinamică, investigație și debugging
+ 
+ În general nu va fi nevoie să dați comenzi de compilare. Fiecare director cuprinde un Makefile pe care îl puteți rula pentru a compila în mod automat fișierele cod sursă limbaj de asamblare sau C.
+
+ ### **1. Iterarea print-un vector de întregi**
+
+ Veți rezolva exercițiul plecând de la fișierul **ex1.c** aflat în directorul **ex1**.
+
+ Se dă urmatoarea bucată de cod în C:
+
+ ```c
+ #include <stdio.h>
+ 
+int main() {
+    int v[] = {0xCAFEBABE, 0xDEADBEEF, 0x0B00B135, 0xBAADF00D, 0xDEADC0DE};
+ 
+    return 0;
+}
+```
+
+Afișați adresele elementelor din vectorul **v** împreună cu valorile de la acestea. Parcurgeți, pe rând, adresele din **v** octet cu octet, din doi în doi, respectiv din patru în patru octeți.
+
+> **TIP:** Puteți parcurge memoria octet cu octet începând de la o anumită adresă folosind un pointer de tipul **unsigned char \*** (pentru că tipul char este reprezentat pe un octet).
+>```c
+>unsigned char *char_ptr = &v;
+>```
+> Pentru afișarea adresei, respectiv a valorii puteți folosi:
+>```c
+>printf("%p -> 0x%x\n", char_ptr, *char_ptr);
+>```
+
+### **2. Ștergerea primei apariții a unui pattern dintr-un șir de caractere**
+
+Veți rezolva exercițiul plecând de la fișierul **ex2.c** aflat în directorul **ex2**.
+
+Dându-se un șir de caractere și un pattern să se implementeze funcția **delete_first(char \*s, char \*pattern)** care întoarce șirul obținut prin ștergerea primei apariții a pattern-ului în s.
+
+> **NOTE:** Pentru s = "Ana are mere" si pattern = "re" se va returna șirul "Ana a mere".
+
+> **IMPORTANT:**
+> Atenție
+> ```c
+> char* s = "Ana are mere" se alocă în .rodata;
+> char s[] = "Ana are mere" se alocă în .text dacă declarația e într-o funcție;
+> char s[] = "Ana are mere" se alocă în .data dacă declarația e în afara funcției;
+> ```
+
+### **3. Pixels**
+
+Veți rezolva exercițiul plecând de la fișierul **ex3.c** aflat în directorul **ex3.**
+
+Se consideră structura unui pixel și a unei imagini descrise în fișierul pixel.h:
+
+```c
+typedef struct Pixel {
+    unsigned char R;
+    unsigned char G;
+    unsigned char B;
+} Pixel;
+ 
+typedef struct Picture {
+    int height;    
+    int width;        
+    Pixel **pix_array;
+} Picture;
+```
+
+Să se implementeze:
+
+1. Funcția **reversePic(Picture \*pic)** care primește ca parametru un Picture și întoarce imaginea răsturnată. Prin imagine răsturnată se înțelege inversarea liniilor matricei pix_array din structura lui Picture.
+2. Functia **colorToGray(Picture \*pic)** care primește ca parametru un Picture și întoarce noua imagine prin convertirea fiecarui pixel la valoarea sa grayscale. Valoarea grayscale a unui pixel se calculează după urmatoarea formulă:
+
+```c
+	p.r = 0.3 * p.r;
```

Comment:
Aici este greșită indentarea în fișierul generat. Probabil ar trebui aliniate la stânga complet.